### ISSUE TITLE
Cut Kubeflow 1.0 RC

### ISSUE THREAD

**Issue by jlewi** [kubeflow/kubeflow#4685]
/kind process

Tracking bug for the Kubeflow 1.0 initial RC.

@richardsliu Has cut a 1.0 branch for kubeflow/manifests
@jlewi Has cut a 1.0 branch for kubeflow/kubeflow

We should begin setting up tests etc... for those branches.

If need be we can rebase those branches on master if there is significant changes that need to be pulled in.

Links:

* Demo script  - http://bit.ly/demo-v1-0
* [Kanban Board For 1.0](https://github.com/orgs/kubeflow/projects/25?card_filter_query=label%3Apriority%2Fp0)

**Comment by cliveseldon** [comment]
Can we get https://github.com/kubeflow/manifests/pull/717 into RC? @jlewi @richardsliu

**Comment by jlewi** [comment]
Here are some to do items

- [ ] We need to get the auto-bot PRs updating docker images on the v1,0-branch merged; https://github.com/kubeflow/manifests/pulls/kubeflow-bot
- [ ] We need to setup auto-deployment from the v1.0-branch
- [ ] We need to get application resources on the v1.0-branch updated
- [ ] KFDefs on the v1.0-branch need to have the manifests revision they point to updated to point to the v1-branch
- [ ] KFDefs on the v1.0-branch should probably be renamed to "*.1.0.0.yaml"

**Comment by jlewi** [comment]
For setting up the kubeflow/manifests v1.0-branch We need to do the following

[ ] Create versioned KFDef specs that point to the v1.0-branch for manifests
[ ] Bump the Application versions on the v1.0-branch

I created kubeflow/manifests#778 to make it simpler to maintained the versioned KFDef YAML files by generating them from kustomize.

Once that is merged we need to cherry pick that onto the release branch.

In that PR I only modified the GCP configs; I didn't want to modify the other configs.

@krishnadurai @Jeffwan @yanniszark @animeshsingh  Do you want to use the same approach for your configs?

**Comment by jlewi** [comment]
Pending cherry-picks

- [ ] https://github.com/kubeflow/manifests/pull/814 - KFDef v1.0.0 specs

I made a copy of the demo script
http://bit.ly/demo-v1-0

**Comment by jlewi** [comment]
Here's the current status

* A kfctl RC is available at https://github.com/kubeflow/kfctl/releases
  * A new RC will be released shortly with updated fixes
* 1.0 KFDef specs for GCP are available in https://github.com/kubeflow/manifests/tree/v1.0-branch/kfdef
   * 1.0 KFDef specs for other platforms are pending in kubeflow/manifests#826
* https://master.kubeflow.org/ - now shows docs for master
  * www.kubeflow.org is directing to a stable 0.7 version of the docs
  * Initial PR to update the docs with 1.0 instructions is pending in kubeflow/website#1583

With the above people should have everything needed to begin testing the demo script: http://bit.ly/demo-v1-0

My suggestion would be 

1. Platform (KFDef owners) do an initial test of the demo script with the RC
1. If everything looks good we email kubeflow-discuss asking others to try it out

I would like to finalize the RC this week or next at the latest.

**Comment by jlewi** [comment]
I think we are super close to finalizing the RC.

* kubeflow/kubeflow#4705 Jupyter UI issue is the only remaining bug
* kubeflow/manifests#834 is a fit and finish issue

**Comment by jlewi** [comment]
I flipped the docs.
www.kubeflow.org now points to master and the 1.0 RC docs.

**Comment by Jeffwan** [comment]
@jlewi 

Sorry for late. We have a few aws specific changes want to be merged and cherry-picked. Can you help take a look? 

- https://github.com/kubeflow/manifests/pull/890/files
- https://github.com/kubeflow/kubeflow/pull/4751
- https://github.com/kubeflow/kubeflow/pull/4747
- https://github.com/kubeflow/kfctl/pull/225
- https://github.com/kubeflow/pipelines/pull/2080

**Comment by jlewi** [comment]
@Jeffwan If the code in question only affects AWS I will defer to you about whether to cherry-pick it. For anything that affects non-AWS as well I would have to take a closer look.

**Comment by jlewi** [comment]
@Jeffwan I took a look at all of the issues. For pipelines which version of pipelines would you need and is it on master yet?

**Comment by Jeffwan** [comment]
@jlewi Thanks! Above Pipeline change is already on master and I hope we can merge back to v1.0-branch. There're two more PRs to add external mysql and object storage overlay and I will check in later tonight.

**Comment by jlewi** [comment]
@Jeffwan how critical are the KFP changes? What breaks without it? Could we possibly hold off for a 1.0.1 release?

**Comment by Jeffwan** [comment]
@jlewi  It's ok to hold off for 1.0.1 for pipeline changes.

**Comment by jlewi** [comment]
The 1.0 RC is looking pretty good in my oppinion.

I think we are waiting on 3 cherry-picks to land in the 1.0 branch

For AWS

* kubeflow/manifests#909 
* kubeflow/manifests@908 

For GCP

* kubeflow/manifests#912

@Jeffwan can we get those PRs merged tomorrow so that we can let it bake over the weekend and the finalize it on Monday.

**Comment by jlewi** [comment]
@krishnadurai @animeshsingh @yanniszark @Jeffwan 

On the GCP side, I think on Tuesday 02/18. we will pin our kfctl_gcp_*.v1.0.0.yaml files to a specific commit of kubeflow/manifests. 

This way we have a stable, repeatable set of manifests. Any future bug fixes would then go into a patch release v1.0.1.yaml. 

If your on board with that then we can pin all of the manifests at once. Otherwise we will just pin the GCP configs and the other configs can track master HEAD.

My suggestion would be that until we pin the v1.0.0.yaml manifests on the v1.0-branch we merge PRs onto the 1.0 branch sparingly. I would propose the following criterion

1. The PRs fix a P0 bug 
1. The PRs should be specific to a particular distribution of Kubeflow to minimize risk to other distributions

The only PRs I'm aware of that still need to be merged are @Jeffwan's outstanding PRs to fix some auth issues for AWS.

**Comment by Jeffwan** [comment]
> @Jeffwan can we get those PRs merged tomorrow so that we can let it bake over the weekend and the finalize it on Monday.

Sure. Could you help approve https://github.com/kubeflow/manifests/pull/916 and https://github.com/kubeflow/manifests/pull/918 ? These two are automatic cherry-pick PRs.

For 1.0.1,we will still use v1.0-branch, could we backport some feature and improvements? Most of the changes are aws specific. I will try to minimize risk and not touch other code bases

**Comment by jlewi** [comment]
@Jeffwan both PRs should be LGTM.

@Jeffwan once we pin the 1.0.0 configs I think its fine to backport changes onto the 1.0.0 branch because that shouldn't affect the 1.0.0 configs (provided we stop regenerating them from the base).

That said we should probably try to stick to semantic versioning and now use patch releases for major functionality. changes.

**Comment by Jeffwan** [comment]
@jlewi Thanks! There's no release blocking issue on AWS side. I will get all PRs ready on master and backport to v1.0-branch after we pin the configs. If there's no major changes on other components, I think patch release with aws features changes fine. I will get changelog ready in case people get confused.

**Comment by jlewi** [comment]
I think we are ready to finalize the release. Remaining steps

1. Publish kfctl 1.0
1. Update docs to use it

**Comment by Jeffwan** [comment]
Any release plan for 1.0.1 or 1.1.0? For kfctl, AWS has two major changes https://github.com/kubeflow/kfctl/pull/254 and https://github.com/kubeflow/kfctl/pull/225. They are all AWS specific changes, it should be very safe to merge, I hope we can cherry-pick back to v1.0-branch before we cut kfctl 1.0, then we don't have to wait for 1.0.1 and update docs one more time.

**Comment by jlewi** [comment]
@Jeffwan Kubeflow 1.0.0 is done. If you need a patch release of kfctl please open an issue in kubeflow/kfctl to plan and track it.

**Comment by Jeffwan** [comment]
Sounds good. Then I will create a separate issue to track my request.
